fluent-syntax/src/parser.js

```diff
@@ -599,37 +590,44 @@ export default class FluentParser {
   getEscapeSequence(ps) {
     const next = ps.currentChar;
 
-    if (next === "\\" || next === "\"") {
-      ps.next();
-      return [`\\${next}`, next];
+    switch (next) {
+      case "\\":
+      case "\"":
+        ps.next();
+        return [`\\${next}`, next];
+      case "u":
+        return this.getUnicodeEscapeSequence(ps, next, 4);
+      case "U":
+        return this.getUnicodeEscapeSequence(ps, next, 6);
+      default:
+        throw new ParseError("E0025", next);
     }
+  }
 
-    if (next === "u") {
-      let sequence = "";
-      ps.next();
-
-      for (let i = 0; i < 4; i++) {
-        const ch = ps.takeHexDigit();
+  getUnicodeEscapeSequence(ps, u, digits) {
+    ps.expectChar(u);
```

Comment:
We already checked this in the switch statement, I'd just ps.next() here, and drop the `u` argument.

Reply:
I'd prefer to keep `expectChar`. In fact, I'm planning a clean up after 0.8 where we get rid of most of `ps.next()`. I always seem to forget what it's supposed to consume, something `expectChar` is very explicit about.

Reply:
Oh, and we need the `u` arg for the error message. That's the real reason why it's there.